runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -2664,23 +2664,35 @@ remoteCompile(
              TR_VerboseLog::writeLineLocked(TR_Vlog_FAILURE, "Interrupting remote compilation (interruptReason %u) in remoteCompile of %s @ %s",
                  interruptReason, comp->signature(), comp->getHotnessName());
 
-         // Inform server that the compilation has been interrupted 
+         // Inform server that the compilation has been interrupted
          client->writeError(JITServer::MessageType::compilationInterrupted, 0 /* placeholder */);
          comp->failCompilation<TR::CompilationInterrupted>("Compilation interrupted in remoteCompile");
          }
 
-      while(!handleServerMessage(client, compiler->fej9vm()));
-      auto recv = client->getRecvData<uint32_t, std::string, std::string, CHTableCommitData, std::vector<TR_OpaqueClassBlock*>,
+      JITServer::MessageType response;
+      while(!handleServerMessage(client, compiler->fej9vm(), response));
+
+      if (JITServer::MessageType::compilationEndCode == response)
+         {
+         auto recv = client->getRecvData<uint32_t, std::string, std::string, CHTableCommitData, std::vector<TR_OpaqueClassBlock*>,
                                       std::string, std::string, std::vector<TR_ResolvedJ9Method*>, TR_OptimizationPlan>();
-      statusCode = std::get<0>(recv);
-      codeCacheStr = std::get<1>(recv);
-      dataCacheStr = std::get<2>(recv);
-      chTableData = std::get<3>(recv);
-      classesThatShouldNotBeNewlyExtended = std::get<4>(recv);
-      logFileStr = std::get<5>(recv);
-      svmSymbolToIdStr = std::get<6>(recv);
-      resolvedMirrorMethodsPersistIPInfo = std::get<7>(recv);
-      modifiedOptPlan = std::get<8>(recv);
+         statusCode = std::get<0>(recv);
+         codeCacheStr = std::get<1>(recv);
+         dataCacheStr = std::get<2>(recv);
+         chTableData = std::get<3>(recv);
+         classesThatShouldNotBeNewlyExtended = std::get<4>(recv);
+         logFileStr = std::get<5>(recv);
+         svmSymbolToIdStr = std::get<6>(recv);
+         resolvedMirrorMethodsPersistIPInfo = std::get<7>(recv);
+         modifiedOptPlan = std::get<8>(recv);
+         }
+      else
+         {
```

Comment:
Please place an assert here that only a `compilationEndErrorCode` can happen here. This will catch future transgressions.